crystalBox.py

```diff
@@ -307,6 +309,37 @@ def getEquivalents(self, hkl):
         print(sg)
         print(pg)
         print("The equivalents of reflection", hkl, "include the following:", equivalents)    
+    
+    def makeCrystal(self):
+        params = [self.a_orig,
+                  self.b_orig,
+                  self.c_orig,
+                  self.alpha_orig,
+                  self.beta_orig,
+                  self.gamma_orig]
+        #print(params)
+        line = ' '.join(['{}']*6)
+        constants = line.format(*params)
+        scatterers = self.get_scatterers()
+        atom_info = ';'.join([line.format(*s) for s in scatterers])
+        crystal_structure = CrystalStructure(constants,self.HMSymbol, atom_info)
+        
+        return crystal_structure
+                        
+    def cartesianHKL(self,h):
+        v = np.matmul(self.BMatrix,h)
+        return v
+    
+    def getAngle(self,vector1,vector2):
```

Comment:
I was again concerned about validation. If some one accidentally specified non-orthogonal vectors as `vector1` and `vector2` they might get confusing output. 

One idea I had: We could create a new class to hold vectors. This would have an attribute `coordinates` (?) that holds the 3 numbers in the vector. But then we could create other useful attributes e.g. 

* a Boolean orthonormal basis flag
* a label whether vector is horizontal or vertical (I feel the maths could easily get mixed up if this isn't tracked
* a method to transform between the two cases above
* a method to return a normal vector pointing in the same direction (i.e. just calculate magnitude and divide all coords by that). This is often useful when doing maths.
* maybe units? ...